### PR TITLE
Chat 2: Part 8 - types / constants / reducers

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

File: shared/actions/json/chat2.json [hunk 1]
```diff
@@ -0,0 +1,322 @@
+{
```
Comment (chrisnojima) [inline shared/actions/json/chat2.json]: new actions! also note this is json5 so it has comments (this was before danny's description thing)

File: shared/constants/chat2/meta.js [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+import * as RPCTypes from '../types/rpc-gen'
+import * as Types from '../types/chat2'
+import type {_ConversationMeta} from '../types/chat2/meta'
+import {formatTimeForConversationList} from '../../util/timestamp'
+import {globalColors} from '../../styles'
+import {isMobile, isIOS, isAndroid} from '../platform'
+import {parseFolderNameToUsers} from '../../util/kbfs'
+import {toByteArray} from 'base64-js'
+
+const IGNORE_UNTRUSTED_CACHE = false
```
Comment (chrisnojima) [inline shared/constants/chat2/meta.js]: this is just to skip the localMetadata we can get back from loading the untrusted inbox for testing

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/constants/chat2/meta.js [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+export const unverifiedInboxUIItemToConversationMeta = (
+  i: RPCChatTypes.UnverifiedInboxUIItem,
+  username: string
+) => {
+  if (IGNORE_UNTRUSTED_CACHE) {
+    // $FlowIssue flow rightfully complains
+    i.localMetadata = null
+  }
+
+  // Public chats only
```
Comment (buoyad) [inline shared/constants/chat2/meta.js]: should this say `Private`?

File: shared/constants/chat2/meta.js [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    notificationsMobile,
+  } = parseNotificationSettings(notifications)
+  return old
+    .set('notificationsDesktop', notificationsDesktop)
+    .set('notificationsGlobalIgnoreMentions', notificationsGlobalIgnoreMentions)
+    .set('notificationsMobile', notificationsMobile)
+}
+
+export const inboxUIItemToConversationMeta = (i: RPCChatTypes.InboxUIItem) => {
+  // Public chats only
```
Comment (buoyad) [inline shared/constants/chat2/meta.js]: Same here

**Comment by chrisnojima** [comment]
@buoyad bump

File: shared/actions/json/chat2.json [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+    // We get new notification settings
+    "notificationSettingsUpdated": {
+      "conversationIDKey": "Types.ConversationIDKey",
+      "settings": "RPCChatTypes.ConversationNotificationInfo"
+    },
+    // To help debug whats going on w/ the chat store
+    "debugDump": {
+      "conversationIDKey?": "?Types.ConversationIDKey",
+    },
+    // Block a user
```
Comment (mmaxim) [inline shared/actions/json/chat2.json]: Technically this blocks the whole convo

File: shared/constants/chat2/meta.js [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+import {formatTimeForConversationList} from '../../util/timestamp'
+import {globalColors} from '../../styles'
+import {isMobile, isIOS, isAndroid} from '../platform'
+import {parseFolderNameToUsers} from '../../util/kbfs'
+import {toByteArray} from 'base64-js'
+
+const IGNORE_UNTRUSTED_CACHE = false
+IGNORE_UNTRUSTED_CACHE && console.log('aaa NOJIMA skipping locametadata')
+
+const conversationMemberStatusToMembershipType = (m: RPCChatTypes.ConversationMemberStatus) => {
```
Comment (mmaxim) [inline shared/constants/chat2/meta.js]: What is the point of this>

File: shared/constants/chat2/meta.js [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+    // $FlowIssue flow rightfully complains
+    i.localMetadata = null
+  }
+
+  // Private chats only
+  if (i.visibility !== RPCTypes.commonTLFVisibility.private) {
+    return null
+  }
+
+  // We only treat implied adhoc teams as having resetParticipants
```
Comment (mmaxim) [inline shared/constants/chat2/meta.js]: We usually use the term "implicit"

File: shared/constants/chat2/meta.js [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+  const participants = I.OrderedSet(
+    i.localMetadata
+      ? i.localMetadata.writerNames || []
+      : parseFolderNameToUsers(username, i.name).map(ul => ul.username)
+  )
+
+  const channelname =
+    i.membersType === RPCChatTypes.commonConversationMembersType.team && i.localMetadata
+      ? i.localMetadata.channelName
+      : ''
```
Comment (mmaxim) [inline shared/constants/chat2/meta.js]: Maybe something better thank blank? Like a dash or `(loading)`

File: shared/constants/chat2/meta.js [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+    m.set('teamname', meta.teamname || old.teamname)
+  })
+}
+
+const parseNotificationSettings = (notifications: ?RPCChatTypes.ConversationNotificationInfo) => {
+  let notificationsDesktop = 'never'
+  let notificationsGlobalIgnoreMentions = false
+  let notificationsMobile = 'never'
+
+  // Map this weird structure from the daemon to something we want
```
Comment (mmaxim) [inline shared/constants/chat2/meta.js]: I also don't understand this. Why not just add some function to pull this info out of the type we give you?

File: shared/constants/chat2/meta.js (hunk 6 above)
Comment (chrisnojima) [reply shared/constants/chat2/meta.js]: i want more UI specific enums passing around and not plumbing through the core ones. also makes this more readable so we can have strings internally so debugging is easier and logging is easier

File: shared/constants/chat2/meta.js (hunk 8 above)
Comment (chrisnojima) [reply shared/constants/chat2/meta.js]: if the goal is to make it show up like that in the inbox then we should do that there and not in our data layer

File: shared/constants/chat2/meta.js (hunk 9 above)
Comment (chrisnojima) [reply shared/constants/chat2/meta.js]: thats what this function is? i don't understand

File: shared/constants/chat2/meta.js (hunk 8 above)
Comment (mmaxim) [reply shared/constants/chat2/meta.js]: Sounds good, I think they just don't show up at all now.

**Pull request merged**